File: HW5/Calculator.fs
module HW5.Calculator

open HW5
open Result

let Calculate (val1: decimal) operation (val2: decimal) =
        match operation with
        | "+" -> Success (val1 + val2)
        | "-" -> Success (val1 - val2)
        | "*" -> Success (val1 * val2)
        | "/" ->
        if val2 = 0m then
           Fail "Divide by zero"
        else
           Success (val1 +  val2)
        | _ -> Fail "Undefined operation"

File: src/HW8/Services/Calculator.cs
using HW8.Models;

namespace HW8.Services
{
    public class Calculator : ICalculator
    {
        public decimal Calculate(decimal left, string operation, decimal right)
        {
            return operation switch
            {
                "+" => left + right,
                "-" => left - right,
                "*" => left * right,
                "/" => right == 0
                    ? 0
                    : left / right,
                _ => 0
            };
        }
    }
}